src/dx/presence/event_engine/effects/mod.rs

```diff
@@ -184,10 +202,23 @@ impl Effect for PresenceEffect {
 
     async fn run(&self) -> Vec<<Self::Invocation as EffectInvocation>::Event> {
         match self {
-            Self::Heartbeat { input, executor } => {
-                heartbeat::execute(input, 0, None, &self.id(), &None, executor).await
+            Self::Heartbeat {
+                id,
+                input,
+                executor,
+            } => {
+                heartbeat::execute(
+                    input,
+                    0,
+                    None,
+                    id,
+                    &RequestRetryConfiguration::None,
+                    executor,
```

Comment:
Maybe not in this PR but it would be worth to prepare `HeartbeatExecuteParams` to make calling of `heartbeat::execute` more readable.

Reply:
Will do in next PRs.